lib/landing_session.js

```diff
@@ -65,34 +68,44 @@ class LandingSession extends Session {
   }
 
   async downloadAndPatch() {
-    const { cli, req, repo, owner, prid } = this;
+    const { cli, repo, owner, prid, expectedCommitShas } = this;
 
-    // TODO(joyeecheung): restore previously downloaded patches
     cli.startSpinner(`Downloading patch for ${prid}`);
-    const patch = await req.text(
-      `https://github.com/${owner}/${repo}/pull/${prid}.patch`);
-    this.savePatch(patch);
-    cli.stopSpinner(`Downloaded patch to ${this.patchPath}`);
+    await runAsync('git', [
+      'fetch', `https://github.com/${owner}/${repo}.git`,
+      `refs/pull/${prid}/merge`]);
+    const [base, head] = (await runAsync('git', [
+      'rev-parse', 'FETCH_HEAD^1', 'FETCH_HEAD^2'], { captureStdout: true }))
```

Comment:
^1, ^2, etc. in git means "prevous commit", "two commits before this commit", etc. Does this have some special meaning in this case due to us fetching the `/merge` ref (like ^1 gets the head of the merge commit, ^2 gets the base of the merge commit)? If that's the case it might be worth adding a comment explaining what's happening.

Reply:
> like ^1 gets the head of the merge commit, ^2 gets the base of the merge commit

Yes, except it’s the other way around :) I’ve added a comment.